datafusion/functions/src/string/common.rs

```diff
@@ -78,6 +80,19 @@ pub(crate) fn general_trim<T: OffsetSizeTrait>(
         2 => {
             let characters_array = as_generic_string_array::<T>(&args[1])?;
 
+            if characters_array.len() == 1 {
+                if characters_array.is_null(0) {
+                    return Ok(new_null_array(args[0].data_type(), args[0].len()));
```

Comment:
This looks like new behavior for null handling? Do we have existing unit tests for this case or can we add a new test as part of this PR?

Reply:
This is not a new behavior. The reason for this logic `characters_array.is_null(0)` is because initially, a test did not pass, and the error was as follows:
```shell
...

External error: query result mismatch:
[SQL] SELECT btrim(' xyxtrimyyx ', NULL)
[Diff] (-expected|+actual)
-   NULL
+    xyxtrimyyx 
at test_files/expr.slt:373

...

note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: test failed, to rerun pass `-p datafusion-sqllogictest --test sqllogictests`
```

Reply:
To provide additional context, this logic is consistent with the `_ => None` here: 

```rust
            let result = string_array
                .iter()
                .zip(characters_array.iter())
                .map(|(string, characters)| match (string, characters) {
                    (Some(string), Some(characters)) => Some(func(string, characters)),
                    _ => None, // If characters is null, append None.
                })
                .collect::<GenericStringArray<T>>();
```

Reply:
Thanks for the clarification @JasonLi-cn